pkg/util/functions/functions.go

```diff
@@ -0,0 +1,38 @@
+package functions
+
+// RunSequentially executes a series of functions sequentially. Each function returns a boolean
+// indicating if the function was successfully, and an error indicating if something went wrong.
+// if any function returns an error, an early exit happens. The first parameter indicates if the functions
+// should run in the order provided. A value of true indicates they should run in reverse.
+func RunSequentially(reversed bool, funcs ...func() (bool, error)) (bool, error) {
+	if !reversed {
+		return runInOrder(funcs...)
+	}
+	return runReversed(funcs...)
+}
+
+func runInOrder(funcs ...func() (bool, error)) (bool, error) {
+	for _, fn := range funcs {
+		ready, err := fn()
+		if err != nil {
+			return ready, err
```

Comment:
[micro-nit] Can we find a better name than `ready`? This package is quite generic and `ready` seems to be more suited for a specific case.